pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -127,6 +127,12 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		returnError = errors.Join(returnError, common.ValidateRules(rules, true,
 			nsrPath.Child(index)))
 	}
+	// Validate fleet workspace rules
+	if newGR.InheritedFleetWorkspacePermissions.ResourceRules != nil {
```

Comment:
It has the `omitempty` json tag, which will set the slice to nil if empty. However, this tag works just if the resource is encoded to json. There are some cases where this encoding might not happen (e.g. kubectl). 
I added this check, and a unit test. Please notice I had to add the raw json in the unit test because the slice is converted to nil when encoding to json